src/core/tranche/manager.sol

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+import "../lightswitch.sol";
+
+contract PileLike {
+    function want() public returns (int);
+}
+
+contract OperatorLike {}
+
+contract DistributorLike {
+    function balance() public;
+}
+
+// TrancheManager
+// Keeps track of the tranches. Manages the interfacing between the tranche side and borrower side of the contracts.
+contract TrancheManager is Switchable, DSNote {
+
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+
+    // --- Data ---
+    DistributorLike public distributor;
+    PileLike public pile;
+
+    // --- Tranches ---
+
+    struct Tranche {
+        uint ratio;
+        OperatorLike operator;
+    }
+
+    Tranche[] public tranches;
+
+    bool public flowThrough;
+    bool public poolClosing;
+
+    constructor (address pile_) public {
+        wards[msg.sender] = 1;
+        pile = PileLike(pile_);
+        flowThrough = false;
+        poolClosing = false;
+    }
+
+    function depend (bytes32 what, address addr) public auth {
+        if (what == "pile") { pile = PileLike(addr); }
+        if (what == "distributor") { distributor = DistributorLike(addr); }
+        else revert();
+    }
+
+    function file(bytes32 what, bool data) public auth {
+        if (what == "flowThrough") { flowThrough = data; }
+        if (what == "poolClosing") { poolClosing = data; }
+    }
+
+    // --- Calls ---
+
+    // TIN tranche should always be added first
+    function addTranche(uint ratio, address operator_) public auth {
```

Comment:
We need a comment for the ratio.
What are we using here, RAD?  For a ratio of 70% is it `7 * 10^26` (0.7) or `70 * 10^27` (70)
```suggestion
    function addTranche(uint ratio, address operator_) public auth {
```

Reply:
70 * 70ˆ27 (RAD)